frontend/src/container/ExplorerOptions/ExplorerOptions.tsx

```diff
@@ -136,39 +192,139 @@ function ExplorerOptions({
 		onMenuItemSelectHandler({
 			key: option.key,
 		});
+		if (ref.current) {
+			ref.current.blur();
+		}
+	};
+
+	const handleClearSelect = (): void => {
+		history.replace(DATASOURCE_VS_ROUTES[sourcepage]);
 	};
 
 	const isQueryUpdated = isStagedQueryUpdated(viewsData?.data?.data, viewKey);
 
+	const {
+		isLoading: isSaveViewLoading,
+		mutateAsync: saveViewAsync,
+	} = useSaveView({
+		viewName: newViewName || '',
+		compositeQuery,
+		sourcePage: sourcepage,
+		extraData: JSON.stringify({ color }),
+	});
+
+	const onSaveHandler = (): void => {
+		saveNewViewHandler({
+			compositeQuery,
+			handlePopOverClose: hideSaveViewModal,
+			extraData: JSON.stringify({ color }),
+			notifications,
+			panelType: panelType || PANEL_TYPES.LIST,
+			redirectWithQueryBuilderData,
+			refetchAllView,
+			saveViewAsync,
+			sourcePage: sourcepage,
+			viewName: newViewName,
+			setNewViewName,
+		});
+	};
+
+	const dropdownStyle: CSSProperties = {
```

Comment:
Let's move it to CSS file.

Reply:
Added comment, will address this while working on view for traces